shell/app-shell/app-shell.js

```diff
@@ -348,13 +358,14 @@ class AppShell extends Xen.Base {
     if (state.user) {
       state.userName = state.user.name;
     }
+    // calculate sharing state
     if (state.config && state.key && state.user) {
       let isProfile = state.user.profiles && state.user.profiles[state.key];
       let isShared = state.user.shares && state.user.shares[state.key];
-      // unpack button states
-      state.profileState = isProfile ? 1 : 0;
-      state.sharedState = isShared ? 1 : 0;
+      state.shareState = isProfile && isShared ? 2 : (isProfile ? 1 : 0);
```

Comment:
For future: I do think constants or an enum or something would be better vs 0, 1, 2. At minimum for code-as-documentation, but also for inevitable debugger scrutiny of values down the line, etc. I know we talked about this briefly on an earlier code review, can discuss more in person, I'd value your insight re: such things in JS land as I am noob in comparison.